src/main/java/net/sargue/time/jsptags/ParseSupport.java

```diff
@@ -38,204 +38,207 @@
+                        if (bodyContent != null && bodyContent.getString() != null) {
+                                input = bodyContent.getString().trim();
+                        }
+                }
+
+                if ((input == null) || input.equals("")) {
+                        if (var != null) {
+                                pageContext.removeAttribute(var, scope);
+                        }
+                        return EVAL_PAGE;
+                }
+
+                // Create formatter
+                DateTimeFormatter formatter;
+                if (pattern != null) {
+                        formatter = DateTimeFormatter.ofPattern(pattern);
+                } else if (style != null) {
+                        formatter = Util.createFormatterForStyle(style);
+                } else {
+                        formatter = Util.createFormatterForStyle("FF");
+                }
+
+                // set formatter locale
+                Locale locale = this.locale;
+                if (locale == null) {
+                        locale = Util.getFormattingLocale(pageContext, true,
+                                DateFormat.getAvailableLocales());
+                }
+                if (locale != null) {
+                        formatter = formatter.withLocale(locale);
+                }
+
+                // set formatter timezone
+                ZoneId tz = this.zoneId;
+                if (tz == null) {
+                        tz = ZoneIdSupport.getZoneId(pageContext, this);
+                }
+                if (tz != null) {
+                        formatter = formatter.withZone(tz);
+                }
+
+                // Parse date
+                TemporalAccessor parsed;
+                try {
+                        parsed = formatter.parse(input, temporalQuery());
+                } catch (DateTimeParseException e) {
+                        throw new JspException(Resources.getMessage(
+                                "PARSE_DATE_PARSE_ERROR", input), e);
+                }
+
+                if (var != null) {
+                        pageContext.setAttribute(var, parsed, scope);
+                } else {
+                        try {
+                                pageContext.getOut().print(parsed);
+                        } catch (IOException ioe) {
+                                throw new JspTagException(ioe.toString(), ioe);
+                        }
+                }
+
+                return EVAL_PAGE;
+        }
+
+        /**
+         * Abstract method to define the query used to format the input with
+         * each specific tag.
+         *
+         * @return the temporal query used to parse the input
+         */
+        protected abstract TemporalQuery<TemporalAccessor> temporalQuery();
+
+        // Releases any resources we may have (or inherit)
+        public void release() {
+                init();
+                super.release();
```

Comment:
Let's also remove this as discussed in the other equivalent change.

Reply:
Done